### ISSUE TITLE
Add target_include_directories PRIVATE to Cmake and remove relative paths from includes

### ISSUE THREAD

**Issue by polusto** [ultravideo/uvgRTP#38]
As @db-tech pointed out in issue #12, it would be better to just add the private target directory and not use relative paths in includes.

**Comment by db-tech** [comment]
Hi,
I've already started to do some refactoring in #13 but was still waiting on #12 to be closed.

Because we now already have 3 issues for the CMake file (#13,  #37, #38), I would suggest to create one issue for recreating the CMake file completely.
There we could collect all the requirements ( including test integration, static code analyses, better install integration...).
Usually I am all for small tasks, but with the Cmake file, that could be tedious.

I am somewhat experienced with CMake so I could help with that or even create a first draft you can review.

**Comment by polusto** [comment]
If I understood correctly, #12 needs the uvgrtp folder added to include(which is a great idea!) to be closed. I can do that after I finish my current task with code duplication.

Are you thinking of fixing #37 and this issue (#38) together with other CMake changes? I can also do some work on those if needed.

A separate issue can be created, especially if you need it to keep track of everything that is about to be improved. A better CMake file would be greatly appreciated in any case. A first draft of the file would be marvelous!

For me in PR:s its much more important that the commits are small, than that the PR:s are small. If there are many small changes that need to be made to one section of code, I can certainly see a reason for a bigger pull request.

**Comment by polusto** [comment]
Merged with #44
